pkgs/development/libraries/boost/generic.nix

```diff
@@ -149,7 +149,7 @@ stdenv.mkDerivation {
   configureScript = "./bootstrap.sh";
   configureFlags = commonConfigureFlags ++ [
     "--with-icu=${icu}"
-    "--with-python=${python}/bin/python"
+    "--with-python=${python}/bin/python${python.majorVersion}"
```

Comment:
Could the python package just be amended to have a `python -> python3` symlink? Or is that unconventional? It just doesn't feel like boost should need to be concerned about python version information like that.